app/src/androidTest/java/org/projectbuendia/client/ui/SnackBarTest.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2015 The Project Buendia Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distrib-
+// uted under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, either express or implied.  See the License for
+// specific language governing permissions and limitations under the License.
+
+package org.projectbuendia.client.ui;
+
+import android.support.test.espresso.NoMatchingViewException;
+import android.view.View;
+import android.widget.Toast;
+
+import junit.framework.Assert;
+
+import org.projectbuendia.client.R;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.RootMatchers.withDecorView;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.IsNot.not;
+
+public class SnackBarTest extends FunctionalTestCase {
+    public void testSimpleMessageSnackBar() {
+        final BaseActivity activity = getActivity();
+        activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                activity.snackBar("This is a SnackBar Test!");
+            }
+        });
+        expectVisibleSoon(viewWithText("This is a SnackBar Test!"));
+    }
+
+    public void testSnackBarWithAction() {
+        final BaseActivity activity = getActivity();
+        activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                activity.snackBar("This is a SnackBar with action Test!", "Action", new View
+                    .OnClickListener() {
+                    @Override public void onClick(View v) {
+                        Toast.makeText(
+                            activity,
+                            "What?",
+                            Toast.LENGTH_LONG
+                        ).show();
+                    }
+                });
+            }
+        });
+        expectVisibleSoon(viewWithText("This is a SnackBar with action Test!"));
+        expectVisible(viewWithId(R.id.snackbar_action));
+        expectVisible(viewThat(hasText("Action")));
+        click(viewWithText("Action"));
+        onView(withText("What?"))
+            .inRoot(withDecorView(not(is(
```

Comment:
Actually, instead of using a `Toast` for this, I think it would be better to use a `Mockito.mock(View.OnClickListener.class)`, and then use `Mockito.verify()` to ensure that the listener was called.